src/Libraries/Microsoft.Extensions.AI/SpeechToText/LoggingSpeechToTextClient.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>A delegating speech to text client that logs speech to text operations to an <see cref="ILogger"/>.</summary>
+/// <para>
+/// The provided implementation of <see cref="ISpeechToTextClient"/> is thread-safe for concurrent use so long as the
+/// <see cref="ILogger"/> employed is also thread-safe for concurrent use.
+/// </para>
+[Experimental("MEAI001")]
+public partial class LoggingSpeechToTextClient : DelegatingSpeechToTextClient
+{
+    /// <summary>An <see cref="ILogger"/> instance used for all logging.</summary>
+    private readonly ILogger _logger;
+
+    /// <summary>The <see cref="JsonSerializerOptions"/> to use for serialization of state written to the logger.</summary>
+    private JsonSerializerOptions _jsonSerializerOptions;
+
+    /// <summary>Initializes a new instance of the <see cref="LoggingSpeechToTextClient"/> class.</summary>
+    /// <param name="innerClient">The underlying <see cref="ISpeechToTextClient"/>.</param>
+    /// <param name="logger">An <see cref="ILogger"/> instance that will be used for all logging.</param>
+    public LoggingSpeechToTextClient(ISpeechToTextClient innerClient, ILogger logger)
+        : base(innerClient)
+    {
+        _logger = Throw.IfNull(logger);
+        _jsonSerializerOptions = AIJsonUtilities.DefaultOptions;
+    }
+
+    /// <summary>Gets or sets JSON serialization options to use when serializing logging data.</summary>
+    public JsonSerializerOptions JsonSerializerOptions
+    {
+        get => _jsonSerializerOptions;
+        set => _jsonSerializerOptions = Throw.IfNull(value);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<SpeechToTextResponse> GetTextAsync(
+        Stream audioSpeechStream, SpeechToTextOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        if (_logger.IsEnabled(LogLevel.Debug))
+        {
+            if (_logger.IsEnabled(LogLevel.Trace))
+            {
+                LogInvokedSensitive(nameof(GetTextAsync), "[audio speech stream]", AsJson(options), AsJson(this.GetService<SpeechToTextClientMetadata>()));
```

Comment:
"[audio speech stream]" doesn't seem valuable to log

Reply:
Removed